pkg/registry/deployment/strategy.go

```diff
@@ -92,6 +92,8 @@ func (deploymentStatusStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newDeployment := obj.(*extensions.Deployment)
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Spec = oldDeployment.Spec
+	newDeployment.Labels = oldDeployment.Labels
```

Comment:
Do we do this for other APIs? We should be consistent, and document this in api conventions.

Does the deployment controller use resourceVersion to ensure it doesn't accidentally clobber labels and annotations?

I agree the controller shouldn't be able to update labels, but I could imagine cases where we'd want it to update annotations in the future.